modules/account-lib/src/coin/algo/utils.ts

```diff
@@ -0,0 +1,611 @@
+import crypto from 'crypto';
+import algosdk from 'algosdk';
+import stellar from 'stellar-sdk';
+import * as hex from '@stablelib/hex';
+import * as nacl from 'tweetnacl';
+import base32 from 'hi-base32';
+import sha512 from 'js-sha512';
+import _ from 'lodash';
+import { isValidEd25519PublicKey, isValidEd25519SecretKey } from '../../utils/crypto';
+import { BaseUtils } from '../baseCoin';
+import { InvalidKey, NotImplementedError, InvalidTransactionError } from '../baseCoin/errors';
+import { EncodedTx, Address, Seed } from './ifaces';
+import { KeyPair } from './keyPair';
+import { SeedEncoding } from './seedEncoding';
+import * as algoNacl from 'algosdk/dist/cjs/src/nacl/naclWrappers';
+import * as encoding from 'algosdk/dist/cjs/src/encoding/encoding';
+
+const ALGORAND_CHECKSUM_BYTE_LENGTH = 4;
+const ALGORAND_ADDRESS_LENGTH = 58;
+const ALGORAND_MINIMUM_FEE = 1000;
+const ALGORAND_SEED_LENGTH = 58;
+const ALGORAND_SEED_BYTE_LENGTH = 36;
+const ALGORAND_TRANSACTION_LENGTH = 52;
+const SEED_BYTES_LENGTH = 32;
+
+/**
+ * Determines whether the string is a composed of hex chars only.
+ *
+ * @param {string} maybe The string to be validated.
+ * @returns {boolean} true if the string consists of only hex characters, otherwise false.
+ */
+function allHexChars(maybe: string): boolean {
+  return maybe.match(/^[0-9a-f]+$/i) !== null;
+}
+
+/**
+ * ConcatArrays takes two array and returns a joint array of both
+ * @param a
+ * @param b
+ * @returns {Uint8Array} [a,b]
+ */
+function concatArrays(a, b) {
```

Comment:
EDIT: I figured out that algosdk will throw if you pass it a `Buffer`, since it checks the `constructor` property of the object being passed in to see if it's really a `Uint8Array`, and `Buffer` doesn't pass this check

No idea why it does that, but ok. Feel free to ignore

Reply:
Still a nit on the input/output types here, but it's not a blocker since this is not part of the module boundary.